### ISSUE TITLE
Android 1.6.1 -> 1.7 upgrade error (status-go/protocol migration error)

### ISSUE THREAD

**Issue by tbenr** [status-im/status-react#11249]
Android 10.
I just upgraded 1.6.1 version to 1.7 from play store.

after upgrade i signed in using fingerprint and i got an empty screen with infinite spinner.

this are corresponding logs:

```
09-30 20:32:35.413 19356 19752 D ReactNativeJS: DEBUG [status-im.ui.screens.routing.core:50] - :on-screen-focus :progress
09-30 20:32:35.423 19356 19753 D StatusModule: clearStorageAPIs
09-30 20:32:35.424 19356 19753 D StatusModule: initKeystore
09-30 20:32:35.473 19356 19752 D ReactNativeJS: DEBUG [status-im.utils.handlers:44] - Handling re-frame event:  :screens/on-will-focus
09-30 20:32:35.475 19356 19752 D ReactNativeJS: DEBUG [status-im.utils.handlers:44] - Handling re-frame event:  :set
09-30 20:32:35.477 19356 19752 D ReactNativeJS: DEBUG [status-im.utils.handlers:44] - Handling re-frame event:  :screens/on-will-focus
09-30 20:32:35.479 19356 19752 D ReactNativeJS: DEBUG [status-im.utils.handlers:44] - Handling re-frame event:  :set
09-30 20:32:35.480 19356 19753 D StatusModule: login
09-30 20:32:35.480 19356 19753 D StatusModule: before migrateKeyStoreDir /data/user/0/im.status.ethereum/no_backup/keystore/0x04de7e932369865d109320c13a7491b737c3925da07d6f58b816ad2d1ea99beb
09-30 20:32:35.481 19356 19753 D StatusModule: Login result: {"error":""}
09-30 20:32:35.719 19356 20095 D StatusModule: Signal event: {"type":"node.started","event":null}
09-30 20:32:35.720 19356 20095 D StatusModule: Signal event: {"type":"node.ready","event":null}
09-30 20:32:35.722 19356 19752 D ReactNativeJS: DEBUG [status-im.utils.handlers:44] - Handling re-frame event:  :signals/signal-received
09-30 20:32:35.723 19356 19752 D ReactNativeJS: DEBUG [status-im.events:575] - :event-str {"type":"node.started","event":null}
09-30 20:32:35.724 19356 19752 D ReactNativeJS: DEBUG [status-im.signals.core:64] - Event  node.started  not handled
09-30 20:32:35.728 19356 19752 D ReactNativeJS: DEBUG [status-im.utils.handlers:44] - Handling re-frame event:  :signals/signal-received
09-30 20:32:35.729 19356 19752 D ReactNativeJS: DEBUG [status-im.events:575] - :event-str {"type":"node.ready","event":null}
09-30 20:32:35.730 19356 19752 D ReactNativeJS: DEBUG [status-im.signals.core:64] - Event  node.ready  not handled
09-30 20:32:35.765 19356 19890 D StatusModule: Signal event: {"type":"discovery.started","event":null}
09-30 20:32:35.768 19356 19752 D ReactNativeJS: DEBUG [status-im.utils.handlers:44] - Handling re-frame event:  :signals/signal-received
09-30 20:32:35.769 19356 19752 D ReactNativeJS: DEBUG [status-im.events:575] - :event-str {"type":"discovery.started","event":null}
09-30 20:32:35.771 19356 19752 D ReactNativeJS: DEBUG [status-im.signals.core:64] - Event  discovery.started  not handled
09-30 20:32:35.945 19356 20097 D StatusModule: Signal event: {"type":"discovery.summary","event":[{"enode":"enode://a8bddfa24e1e92a82609b390766faa56cf7a5eef85b22a2b51e79b333c8aaeec84f7b4267e432edd1cf45b63a3ad0fc7d6c3a16f046aa6bc07ebe50e80b63b8c@178.128.141.249:443","id":"34f766b58ca7f8ea197bfd2ffb377c474acb4b254cff4ad693a5b403e82001df","name":"Statusd/v0.61.2/linux-amd64/go1.13.8","caps":["shh/6","waku/0","waku/1"],"network":{"localAddress":"192.168.0.104:37200","remoteAddress":"178.128.141.249:443","inbound":false,"trusted":false,"static":true},"protocols":{"waku":"unknown"}}]}
09-30 20:32:35.947 19356 19752 D ReactNativeJS: DEBUG [status-im.utils.handlers:44] - Handling re-frame event:  :signals/signal-received
09-30 20:32:35.949 19356 19752 D ReactNativeJS: DEBUG [status-im.events:575] - :event-str {"type":"discovery.summary","event":[{"enode":"enode://a8bddfa24e1e92a82609b390766faa56cf7a5eef85b22a2b51e79b333c8aaeec84f7b4267e432edd1cf45b63a3ad0fc7d6c3a16f046aa6bc07ebe50e80b63b8c@178.128.141.249:443","id":"34f766b58ca7f8ea197bfd2ffb377c474acb4b254cff4ad693a5b403e82001df","name":"Statusd/v0.61.2/linux-amd64/go1.13.8","caps":["shh/6","waku/0","waku/1"],"network":{"localAddress":"192.168.0.104:37200","remoteAddress":"178.128.141.249:443","inbound":false,"trusted":false,"static":true},"protocols":{"waku":"unknown"}}]}
09-30 20:32:36.138 19356 19889 D StatusModule: Signal event: {"type":"discovery.summary","event":[{"enode":"enode://a8bddfa24e1e92a82609b390766faa56cf7a5eef85b22a2b51e79b333c8aaeec84f7b4267e432edd1cf45b63a3ad0fc7d6c3a16f046aa6bc07ebe50e80b63b8c@178.128.141.249:443","id":"34f766b58ca7f8ea197bfd2ffb377c474acb4b254cff4ad693a5b403e82001df","name":"Statusd/v0.61.2/linux-amd64/go1.13.8","caps":["shh/6","waku/0","waku/1"],"network":{"localAddress":"192.168.0.104:37200","remoteAddress":"178.128.141.249:443","inbound":false,"trusted":false,"static":true},"protocols":{"waku":"unknown"}},{"enode":"enode://0624b4a90063923c5cc27d12624b6a49a86dfb3623fcb106801217fdbab95f7617b83fa2468b9ae3de593ff6c1cf556ccf9bc705bfae9cb4625999765127b423@35.222.158.246:443","id":"3a26b784ecaef2ea57bdb4455a2c10bbfa53164ddbb22abed34bf78f4d3064c3","name":"Statusd/v0.61.2/linux-amd64/go1.13.8","caps":["shh/6","waku/0","waku/1"],"network":{"localAddress":"192.168.0.104:39606","remoteAddress":"35.222.158.246:443","inbound":false,"trusted":false,"static":true},"protocols":{"waku":"unknown"}}]}
09-30 20:32:36.144 19356 19752 D ReactNativeJS: DEBUG [status-im.utils.handlers:44] - Handling re-frame event:  :signals/signal-received
09-30 20:32:36.146 19356 19752 D ReactNativeJS: DEBUG [status-im.events:575] - :event-str {"type":"discovery.summary","event":[{"enode":"enode://a8bddfa24e1e92a82609b390766faa56cf7a5eef85b22a2b51e79b333c8aaeec84f7b4267e432edd1cf45b63a3ad0fc7d6c3a16f046aa6bc07ebe50e80b63b8c@178.128.141.249:443","id":"34f766b58ca7f8ea197bfd2ffb377c474acb4b254cff4ad693a5b403e82001df","name":"Statusd/v0.61.2/linux-amd64/go1.13.8","caps":["shh/6","waku/0","waku/1"],"network":{"localAddress":"192.168.0.104:37200","remoteAddress":"178.128.141.249:443","inbound":false,"trusted":false,"static":true},"protocols":{"waku":"unknown"}},{"enode":"enode://0624b4a90063923c5cc27d12624b6a49a86dfb3623fcb106801217fdbab95f7617b83fa2468b9ae3de593ff6c1cf556ccf9bc705bfae9cb4625999765127b423@35.222.158.246:443","id":"3a26b784ecaef2ea57bdb4455a2c10bbfa53164ddbb22abed34bf78f4d3064c3","name":"Statusd/v0.61.2/linux-amd64/go1.13.8","caps":["shh/6","waku/0","waku/1"],"network":{"localAddress":"192.168.0.104:39606","remoteAddress":"35.222.158.246:443","inbound":false,"trusted":false,"static":true},"protocols":{"waku":"unknown"}}]}
09-30 20:32:36.406 19356 20098 D StatusModule: Signal event: {"type":"discovery.summary","event":[{"enode":"enode://a8bddfa24e1e92a82609b390766faa56cf7a5eef85b22a2b51e79b333c8aaeec84f7b4267e432edd1cf45b63a3ad0fc7d6c3a16f046aa6bc07ebe50e80b63b8c@178.128.141.249:443","id":"34f766b58ca7f8ea197bfd2ffb377c474acb4b254cff4ad693a5b403e82001df","name":"Statusd/v0.61.2/linux-amd64/go1.13.8","caps":["shh/6","waku/0","waku/1"],"network":{"localAddress":"192.168.0.104:37200","remoteAddress":"178.128.141.249:443","inbound":false,"trusted":false,"static":true},"protocols":{"waku":"unknown"}},{"enode":"enode://0624b4a90063923c5cc27d12624b6a49a86dfb3623fcb106801217fdbab95f7617b83fa2468b9ae3de593ff6c1cf556ccf9bc705bfae9cb4625999765127b423@35.222.158.246:443","id":"3a26b784ecaef2ea57bdb4455a2c10bbfa53164ddbb22abed34bf78f4d3064c3","name":"Statusd/v0.61.2/linux-amd64/go1.13.8","caps":["shh/6","waku/0","waku/1"],"network":{"localAddress":"192.168.0.104:39606","remoteAddress":"35.222.158.246:443","inbound":false,"trusted":false,"static":true},"protocols":{"waku":"unknown"}},{"enode":"enode://b957e51f41e4abab8382e1ea7229e88c6e18f34672694c6eae389eac22dab8655622bbd4a08192c321416b9becffaab11c8e2b7a5d0813b922aa128b82990dab@47.75.222.178:30305?discport=0","id":"7e9e962838184310f2b6cf21203d50b60bd61277936ebc2c771ab1313d5880c9","name":"Statusd/v0.61.2/linux-amd64/go1.13.8","caps":["shh/6","waku/0","waku/1"],"network":{"localAddress":"192.168.0.104:53780","remoteAddress":"47.75.222.178:30305","inbound":false,"trusted":false,"static":true},"protocols":{"waku":"unknown"}}]}
09-30 20:32:36.410 19356 19752 D ReactNativeJS: DEBUG [status-im.utils.handlers:44] - Handling re-frame event:  :signals/signal-received
09-30 20:32:36.412 19356 19752 D ReactNativeJS: DEBUG [status-im.events:575] - :event-str {"type":"discovery.summary","event":[{"enode":"enode://a8bddfa24e1e92a82609b390766faa56cf7a5eef85b22a2b51e79b333c8aaeec84f7b4267e432edd1cf45b63a3ad0fc7d6c3a16f046aa6bc07ebe50e80b63b8c@178.128.141.249:443","id":"34f766b58ca7f8ea197bfd2ffb377c474acb4b254cff4ad693a5b403e82001df","name":"Statusd/v0.61.2/linux-amd64/go1.13.8","caps":["shh/6","waku/0","waku/1"],"network":{"localAddress":"192.168.0.104:37200","remoteAddress":"178.128.141.249:443","inbound":false,"trusted":false,"static":true},"protocols":{"waku":"unknown"}},{"enode":"enode://0624b4a90063923c5cc27d12624b6a49a86dfb3623fcb106801217fdbab95f7617b83fa2468b9ae3de593ff6c1cf556ccf9bc705bfae9cb4625999765127b423@35.222.158.246:443","id":"3a26b784ecaef2ea57bdb4455a2c10bbfa53164ddbb22abed34bf78f4d3064c3","name":"Statusd/v0.61.2/linux-amd64/go1.13.8","caps":["shh/6","waku/0","waku/1"],"network":{"localAddress":"192.168.0.104:39606","remoteAddress":"35.222.158.246:443","inbound":false,"trusted":false,"static":true},"protocols":{"waku":"unknown"}},{"enode":"enode://b957e51f41e4abab8382e1ea7229e88c6e18f34672694c6eae389eac22dab8655622bbd4a08192c321416b9becffaab11c8e2b7a5d0813b922aa128b82990dab@47.75.222.178:30305?discport=0","id":"7e9e962838184310f2b6cf21203d50b60bd61277936ebc2c771ab1313d5880c9","name":"Statusd/v0.61.2/linux-amd64/go1.13.8","caps":["shh/6","waku/0","waku/1"],"network":{"localAddress":"192.168.0.104:53780","remoteAddress":"47.75.222.178:30305","inbound":false,"trusted":false,"static":true},"protocols":{"waku":"unknown"}}]}
09-30 20:32:36.414 19356 19774 D StatusModule: Signal event: {"type":"discovery.summary","event":[{"enode":"enode://cf5f7a7e64e3b306d1bc16073fba45be3344cb6695b0b616ccc2da66ea35b9f35b3b231c6cf335fdfaba523519659a440752fc2e061d1e5bc4ef33864aac2f19@47.75.221.196:30305?discport=0","id":"3204fe475d0e471da3ca3f8e43cb9e278983cabf85c6314254c951f9a7da4216","name":"Statusd/v0.61.2/linux-amd64/go1.13.8","caps":["shh/6","waku/0","waku/1"],"network":{"localAddress":"192.168.0.104:57724","remoteAddress":"47.75.221.196:30305","inbound":false,"trusted":false,"static":true},"protocols":{"waku":"unknown"}},{"enode":"enode://a8bddfa24e1e92a82609b390766faa56cf7a5eef85b22a2b51e79b333c8aaeec84f7b4267e432edd1cf45b63a3ad0fc7d6c3a16f046aa6bc07ebe50e80b63b8c@178.128.141.249:443","id":"34f766b58ca7f8ea197bfd2ffb377c474acb4b254cff4ad693a5b403e82001df","name":"Statusd/v0.61.2/linux-amd64/go1.13.8","caps":["shh/6","waku/0","waku/1"],"network":{"localAddress":"192.168.0.104:37200","remoteAddress":"178.128.141.249:443","inbound":false,"trusted":false,"static":true},"protocols":{"waku":"unknown"}},{"enode":"enode://0624b4a90063923c5cc27d12624b6a49a86dfb3623fcb106801217fdbab95f7617b83fa2468b9ae3de593ff6c1cf556ccf9bc705bfae9cb4625999765127b423@35.222.158.246:443","id":"3a26b784ecaef2ea57bdb4455a2c10bbfa53164ddbb22abed34bf78f4d3064c3","name":"Statusd/v0.61.2/linux-amd64/go1.13.8","caps":["shh/6","waku/0","waku/1"],"network":{"localAddress":"192.168.0.104:39606","remoteAddress":"35.222.158.246:443","inbound":false,"trusted":false,"static":true},"protocols":{"waku":"unknown"}},{"enode":"enode://b957e51f41e4abab8382e1ea7229e88c6e18f34672694c6eae389eac22dab8655622bbd4a08192c321416b9becffaab11c8e2b7a5d0813b922aa128b82990dab@47.75.222.178:30305?discport=0","id":"7e9e962838184310f2b6cf21203d50b60bd61277936ebc2c771ab1313d5880c9","name":"Statusd/v0.61.2/linux-amd64/go1.13.8","caps":["shh/6","waku/0","waku/1"],"network":{"localAddress":"192.168.0.104:53780","remoteAddress":"47.75.222.178:30305","inbound":false,"trusted":false,"static":true},"protocols":{"waku":"unknown"}}]}
09-30 20:32:36.420 19356 19752 D ReactNativeJS: DEBUG [status-im.utils.handlers:44] - Handling re-frame event:  :signals/signal-received
09-30 20:32:36.422 19356 19752 D ReactNativeJS: DEBUG [status-im.events:575] - :event-str {"type":"discovery.summary","event":[{"enode":"enode://cf5f7a7e64e3b306d1bc16073fba45be3344cb6695b0b616ccc2da66ea35b9f35b3b231c6cf335fdfaba523519659a440752fc2e061d1e5bc4ef33864aac2f19@47.75.221.196:30305?discport=0","id":"3204fe475d0e471da3ca3f8e43cb9e278983cabf85c6314254c951f9a7da4216","name":"Statusd/v0.61.2/linux-amd64/go1.13.8","caps":["shh/6","waku/0","waku/1"],"network":{"localAddress":"192.168.0.104:57724","remoteAddress":"47.75.221.196:30305","inbound":false,"trusted":false,"static":true},"protocols":{"waku":"unknown"}},{"enode":"enode://a8bddfa24e1e92a82609b390766faa56cf7a5eef85b22a2b51e79b333c8aaeec84f7b4267e432edd1cf45b63a3ad0fc7d6c3a16f046aa6bc07ebe50e80b63b8c@178.128.141.249:443","id":"34f766b58ca7f8ea197bfd2ffb377c474acb4b254cff4ad693a5b403e82001df","name":"Statusd/v0.61.2/linux-amd64/go1.13.8","caps":["shh/6","waku/0","waku/1"],"network":{"localAddress":"192.168.0.104:37200","remoteAddress":"178.128.141.249:443","inbound":false,"trusted":false,"static":true},"protocols":{"waku":"unknown"}},{"enode":"enode://0624b4a90063923c5cc27d12624b6a49a86dfb3623fcb106801217fdbab95f7617b83fa2468b9ae3de593ff6c1cf556ccf9bc705bfae9cb4625999765127b423@35.222.158.246:443","id":"3a26b784ecaef2ea57bdb4455a2c10bbfa53164ddbb22abed34bf78f4d3064c3","name":"Statusd/v0.61.2/linux-amd64/go1.13.8","caps":["shh/6","waku/0","waku/1"],"network":{"localAddress":"192.168.0.104:39606","remoteAddress":"35.222.158.246:443","inbound":false,"trusted":false,"static":true},"protocols":{"waku":"unknown"}},{"enode":"enode://b957e51f41e4abab8382e1ea7229e88c6e18f34672694c6eae389eac22dab8655622bbd4a08192c321416b9becffaab11c8e2b7a5d0813b922aa128b82990dab@47.75.222.178:30305?discport=0","id":"7e9e962838184310f2b6cf21203d50b60bd61277936ebc2c771ab1313d5880c9","name":"Statusd/v0.61.2/linux-amd64/go1.13.8","caps":["shh/6","waku/0","waku/1"],"network":{"localAddress":"192.168.0.104:53780","remoteAddress":"47.75.222.178:30305","inbound":false,"trusted":false,"static":true},"protocols":{"waku":"unknown"}}]}
09-30 20:32:46.418 19356 19889 D StatusModule: Signal event: {"type":"discovery.stopped","event":null}
09-30 20:32:46.453 19356 19752 D ReactNativeJS: DEBUG [status-im.utils.handlers:44] - Handling re-frame event:  :signals/signal-received
09-30 20:32:46.456 19356 19752 D ReactNativeJS: DEBUG [status-im.events:575] - :event-str {"type":"discovery.stopped","event":null}
09-30 20:32:46.459 19356 19752 D ReactNativeJS: DEBUG [status-im.signals.core:64] - Event  discovery.stopped  not handled
```

then i killed the app and tried login with password and i got:

![image](https://user-images.githubusercontent.com/15999009/94726670-d7112900-035d-11eb-8770-89d99a162597.png)

corresponding logs:

```
09-30 20:34:21.763 31589 31902 D StatusModule: clearCookies
09-30 20:34:21.763 31589 31901 D ReactNativeJS: DEBUG [status-im.ui.screens.routing.core:50] - :on-screen-focus :progress
09-30 20:34:21.771 31589 31902 D StatusModule: clearStorageAPIs
09-30 20:34:21.776 31589 31902 D StatusModule: initKeystore
09-30 20:34:21.822 31589 31901 D ReactNativeJS: DEBUG [status-im.utils.handlers:44] - Handling re-frame event:  :screens/on-will-focus
09-30 20:34:21.824 31589 31901 D ReactNativeJS: DEBUG [status-im.utils.handlers:44] - Handling re-frame event:  :set
09-30 20:34:21.826 31589 31901 D ReactNativeJS: DEBUG [status-im.utils.handlers:44] - Handling re-frame event:  :screens/on-will-focus
09-30 20:34:21.828 31589 31901 D ReactNativeJS: DEBUG [status-im.utils.handlers:44] - Handling re-frame event:  :set
09-30 20:34:21.830 31589 31902 D StatusModule: login
09-30 20:34:21.830 31589 31902 D StatusModule: before migrateKeyStoreDir /data/user/0/im.status.ethereum/no_backup/keystore/0x04de7e932369865d109320c13a7491b737c3925da07d6f58b816ad2d1ea99beb
09-30 20:34:21.832 31589 31902 D StatusModule: Login result: {"error":""}
09-30 20:34:22.145 31589 31957 D StatusModule: Signal event: {"type":"node.started","event":null}
09-30 20:34:22.146 31589 31957 D StatusModule: Signal event: {"type":"node.ready","event":null}
09-30 20:34:22.174 31589 31901 D ReactNativeJS: DEBUG [status-im.utils.handlers:44] - Handling re-frame event:  :signals/signal-received
09-30 20:34:22.190 31589 31901 D ReactNativeJS: DEBUG [status-im.events:575] - :event-str {"type":"node.started","event":null}
09-30 20:34:22.191 31589 31901 D ReactNativeJS: DEBUG [status-im.signals.core:64] - Event  node.started  not handled
09-30 20:34:22.193 31589 31901 D ReactNativeJS: DEBUG [status-im.utils.handlers:44] - Handling re-frame event:  :signals/signal-received
09-30 20:34:22.194 31589 31901 D ReactNativeJS: DEBUG [status-im.events:575] - :event-str {"type":"node.ready","event":null}
09-30 20:34:22.195 31589 31901 D ReactNativeJS: DEBUG [status-im.signals.core:64] - Event  node.ready  not handled
09-30 20:34:22.222 31589 32395 D StatusModule: Signal event: {"type":"discovery.started","event":null}
09-30 20:34:22.226 31589 31901 D ReactNativeJS: DEBUG [status-im.utils.handlers:44] - Handling re-frame event:  :signals/signal-received
09-30 20:34:22.227 31589 31901 D ReactNativeJS: DEBUG [status-im.events:575] - :event-str {"type":"discovery.started","event":null}
09-30 20:34:22.228 31589 31901 D ReactNativeJS: DEBUG [status-im.signals.core:64] - Event  discovery.started  not handled
09-30 20:34:22.315 31589 31920 D StatusModule: Signal event: {"type":"discovery.summary","event":[{"enode":"enode://a8bddfa24e1e92a82609b390766faa56cf7a5eef85b22a2b51e79b333c8aaeec84f7b4267e432edd1cf45b63a3ad0fc7d6c3a16f046aa6bc07ebe50e80b63b8c@178.128.141.249:443","id":"34f766b58ca7f8ea197bfd2ffb377c474acb4b254cff4ad693a5b403e82001df","name":"Statusd/v0.61.2/linux-amd64/go1.13.8","caps":["shh/6","waku/0","waku/1"],"network":{"localAddress":"192.168.0.104:37340","remoteAddress":"178.128.141.249:443","inbound":false,"trusted":false,"static":true},"protocols":{"waku":"unknown"}}]}
09-30 20:34:22.318 31589 31901 D ReactNativeJS: DEBUG [status-im.utils.handlers:44] - Handling re-frame event:  :signals/signal-received
09-30 20:34:22.319 31589 31901 D ReactNativeJS: DEBUG [status-im.events:575] - :event-str {"type":"discovery.summary","event":[{"enode":"enode://a8bddfa24e1e92a82609b390766faa56cf7a5eef85b22a2b51e79b333c8aaeec84f7b4267e432edd1cf45b63a3ad0fc7d6c3a16f046aa6bc07ebe50e80b63b8c@178.128.141.249:443","id":"34f766b58ca7f8ea197bfd2ffb377c474acb4b254cff4ad693a5b403e82001df","name":"Statusd/v0.61.2/linux-amd64/go1.13.8","caps":["shh/6","waku/0","waku/1"],"network":{"localAddress":"192.168.0.104:37340","remoteAddress":"178.128.141.249:443","inbound":false,"trusted":false,"static":true},"protocols":{"waku":"unknown"}}]}
09-30 20:34:22.500 31589 31956 D StatusModule: Signal event: {"type":"discovery.stopped","event":null}
09-30 20:34:22.502 31589 31956 D StatusModule: Signal event: {"type":"node.stopped","event":null}
09-30 20:34:22.502 31589 31956 D StatusModule: Signal event: {"type":"node.login","event":{"error":"failed to apply migrations: failed to apply status-go/protocol migrations: failed to migrate: Dirty database version 1599720851. Fix and force version."}}
09-30 20:34:22.508 31589 31901 D ReactNativeJS: DEBUG [status-im.utils.handlers:44] - Handling re-frame event:  :signals/signal-received
09-30 20:34:22.509 31589 31901 D ReactNativeJS: DEBUG [status-im.events:575] - :event-str {"type":"discovery.stopped","event":null}
09-30 20:34:22.510 31589 31901 D ReactNativeJS: DEBUG [status-im.signals.core:64] - Event  discovery.stopped  not handled
09-30 20:34:22.512 31589 31901 D ReactNativeJS: DEBUG [status-im.utils.handlers:44] - Handling re-frame event:  :signals/signal-received
09-30 20:34:22.513 31589 31901 D ReactNativeJS: DEBUG [status-im.events:575] - :event-str {"type":"node.stopped","event":null}
09-30 20:34:22.514 31589 31901 D ReactNativeJS: DEBUG [status-im.signals.core:64] - Event  node.stopped  not handled
09-30 20:34:22.516 31589 31901 D ReactNativeJS: DEBUG [status-im.utils.handlers:44] - Handling re-frame event:  :signals/signal-received
09-30 20:34:22.517 31589 31901 D ReactNativeJS: DEBUG [status-im.events:575] - :event-str {"type":"node.login","event":{"error":"failed to apply migrations: failed to apply status-go/protocol migrations: failed to migrate: Dirty database version 1599720851. Fix and force version."}}
09-30 20:34:22.519 31589 31901 D ReactNativeJS: DEBUG [status-im.signals.core:14] - [signals] status-node-started error failed to apply migrations: failed to apply status-go/protocol migrations: failed to migrate: Dirty database version 1599720851. Fix and force version.
09-30 20:34:43.486 31589 31901 D ReactNativeJS: DEBUG [status-im.utils.handlers:44] - Handling re-frame event:  :shake-event
09-30 20:34:43.655 31589 31901 D ReactNativeJS: DEBUG [status-im.utils.handlers:44] - Handling re-frame event:  :shake-event
09-30 20:34:43.657 31589 31901 D ReactNativeJS: DEBUG [status-im.utils.handlers:44] - Handling re-frame event:  :shake-event```

**Comment by hesterbruikman** [comment]
Thanks for reporting @tbenr

**Comment by flexsurfer** [comment]
thanks @tbenr are you sure you had 1.6.1 probably you installed nightly ?

**Comment by churik** [comment]
No luck to reproduce so far:
- tried with several days of spamming history (thousands of messages) in 3 different channels
- with fingerprint login enabled.
Investigating.

**Comment by tbenr** [comment]
@flexsurfer no it was a 1.6.1 from playstore

@churik i was in many public channels, and it was an installation that came through many upgrades. hope it helps.
i waited for many seconds (around 1min? maybe less...) before killing the app.

**Comment by cammellos** [comment]
The issues is probably because the migration was interrupted. 
Our migration library does not run migrations in a transaction, so if that happens the db is marked as "dirty" and requires manual intervention: https://github.com/golang-migrate/migrate/issues/282
The likelyexplanation steps:
```
once you open the account the first time after you installed 1.7
it will migrate the database
during the login screen 
and how long it takes depends on how many messages you have
with me it took around 15 seconds the first time
but if you have many messages it might take longer, 30/45 even a minute
with biometric enabled, it looks like the app is hanging, as you are presented with a blank page with a spinner, while if you don't is clearer as you are on the login page
so if the app is killed during the process, it will result in a dirty migration table
```

**Comment by cammellos** [comment]
We probably want to address this at the UX level, and at the very least provide a build to "fix" the issue (not through playstore), also worth maybe looking into making migrations run in a transaction if that's feasible, as we already have our own fork of golang migrate, so that if whatever reason the migration fails, no changes will be made and the database will be restored to pre-migration state.

**Comment by cammellos** [comment]
Another thing we want to do if we can't run all the migrations in a single transaction (that'd be ideal), is to run each single migration in a transaction, so that we can safely "force" the version in special builds that will fix the issue.

**Comment by churik** [comment]
So with history 10-30 Sept (couldn't fetch more because can't scroll to the beginning) in `status` and `support` channels, several 1-1 chats and 3 blocked users:
- login takes around 8s with fingerprint enabled

Please tell me if I can help somehow more, because I don't see another way now to reproduce the issue.

**Comment by Jas0n-F** [comment]
Exactly what @cammellos described happened to me. 

I killed the app at approx 30 sec after spinner started. reopened and saw the same error message.
I had a number of messages, biometric enabled.

**Comment by cammellos** [comment]
thanks for the report @Jas0n-F , I am testing a fix, it might be a build you need to manually install as if not many users are affected we might not distribute through the play store. I will keep you updated.

**Comment by Jas0n-F** [comment]
No problem. For the record, I am running:
One Plus 5T
Android 8.10 / Lineage OS 15.1

**Comment by churik** [comment]
Thank you guys, I did manage to replicate it and it is not related to biometric-enabled. All is about the interruption of process  - testing fix currently

**Comment by cammellos** [comment]
This build should fix the issue if anyone wants to try it: 

http://status-im.ams3.digitaloceanspaces.com/StatusIm-201001-085711-427368-release-universal.apk

**Comment by churik** [comment]
It was tested and works for me https://github.com/status-im/status-go/pull/2052#issuecomment-702052989

**Comment by flexsurfer** [comment]
hey @tbenr 1.7.1 is out, could you try it ?

**Comment by tbenr** [comment]
@flexsurfer unfortunately I uninstalled\restored my account on a fresh 1.7 before opening this issue.. so i cant retest the same situation anymore.. but i upgraded 1.7 to 1.7.1 and went ok.

**Comment by flexsurfer** [comment]
ok, thanks

**Comment by flexsurfer** [comment]
hey @Jas0n-F how about you ? :)

**Comment by cammellos** [comment]
This has now been fixed with 1.7.1 hotfix, will close for now, sorry for the inconvenience and thanks for the your patience.
